src/gtc/passes/gtir_upcaster.py

```diff
@@ -58,7 +58,7 @@ def _numpy_ufunc_upcasting_rule(*dtypes, ufunc: np.ufunc):
         inputs, output = t.split("->")
         assert len(inputs) == len(dtypes)
         if all(
-            np.can_cast(data_type_to_typestr(arg_dtype), cand_typestr)
+            arg_dtype <= typestr_to_data_type(np.dtype(cand_typestr).str)
             for arg_dtype, cand_typestr in zip(dtypes, inputs)
         ):
```

Comment:
Does this change mean that bools are upcast to integers/floats? If so, do we want that behavior?

Reply:
While it would be allowed here, can_cast allows it too, but it is caught elsewhere in the pipeline. (BinaryOp with ArithmeticOperator does not allow boolean types.)

Reply:
Great. Thanks for jogging my memory on that!